scylla/src/transport/large_batch_statements_test.rs

```diff
@@ -0,0 +1,106 @@
+use bcs::serialize_into;
+use scylla_cql::errors::{BadQuery, QueryError};
+
+use crate::batch::BatchType;
+use crate::query::Query;
+use crate::{
+    batch::Batch,
+    prepared_statement::PreparedStatement,
+    test_utils::{create_new_session_builder, unique_keyspace_name},
+    IntoTypedRows, QueryResult, Session,
+};
+
+#[tokio::test]
+async fn test_large_batch_statements() {
+    let mut session = create_new_session_builder().build().await.unwrap();
+    let ks = unique_keyspace_name();
+    session = create_test_session(session, &ks).await;
+
+    let max_number_of_queries = u16::MAX as usize;
+    write_batch(&session, max_number_of_queries).await;
+
+    let key_prefix = vec![0];
+    let keys = find_keys_by_prefix(&session, key_prefix.clone()).await;
+    assert_eq!(keys.len(), max_number_of_queries);
+
+    let too_many_queries = u16::MAX as usize + 1;
+
+    let err = write_batch(&session, too_many_queries).await;
+
+    assert!(err.is_err());
```

Comment:
Please make sure that the error is actually `TooManyQueriesInBatchStatement`.
I think this can be done using `assert_matches!`.

Reply:
Thanks, knew there was a way to do this without having to implement `PartialEq` on everything 😢